griptape/tasks/prompt_task.py

```diff
@@ -79,7 +79,7 @@ def prompt_stack(self) -> PromptStack:
         if self.output:
             stack.add_assistant_message(self.output)
 
-        if memory is not None:
+        if memory is not None and isinstance(memory, BaseConversationMemory):
```

Comment:
nit: can this be 
```suggestion
        if isinstance(memory, BaseConversationMemory):
```
or 
```suggestion
        if memory is not None and memory is not NothingType:
```
?

Reply:
While it works, I think the first one kind of hides some behavior. Updated to a variation on the second.